pytorch_lightning/trainer/trainer.py

```diff
@@ -646,7 +647,7 @@ def _setup_on_init(self, num_sanity_val_steps: int) -> None:
         self.should_stop = False
         self.state = TrainerState()
         self.num_training_batches = float("inf")
-        self.train_dataloader = None
+        self.train_dataloader: DataLoader = None
```

Comment:
Should it be `Optional` because `None` is assigned to it?
```suggestion
        self.train_dataloader: Optional[DataLoader] = None
```

Reply:
@akihironitta this leads to the mypy error 
`pytorch_lightning/loops/fit_loop.py:226: error: Argument 1 to "process_dataloader" of "Strategy" has incompatible type "Optional[DataLoader[Any]]"; expected "DataLoader[Any]"  [arg-type]`

Not sure if I should change the types in signature of `process_dataloader`, or just put this line back to: `self.train_dataloader: DataLoader = None`

Reply:
@daniellepintz `self.train_dataloader: DataLoader = None` is the wrong type for sure, but I think we can revert it here and handle it in #7037 because this PR is not about typing, and there are heaps of mypy errors in the file anyway (which is currently ignored as configured in `pyproject.toml`).

Reply:
sounds good! thanks

Reply:
I'm not convinced by this. This annotation is wrong. The correct thing would be to add a check or assertion in the fit_loop.

#7037 is way too large so this annotation will likely get forgotten and can give us a false sense of security because mypy will no longer know it's optional.

Reply:
That makes sense, thanks. Updated